Sources/MapLibreSwiftUI/MapViewCoordinator.swift

```diff
@@ -0,0 +1,194 @@
+import Foundation
+import MapLibre
+import MapLibreSwiftDSL
+
+public class MapViewCoordinator: NSObject {
+    
+    // This must be weak, the UIViewRepresentable owns the MLNMapView.
+    weak var mapView: MLNMapView?
```

Comment:
Something to discuss: this introduces an interesting split in the API now.

Up until this point, the coordinator was carefully designed around not needing to store a reference to the map view; this seems like generally good practice. I believe we're departing from that now as a result of adding gesture recognizers which are pretty old school selector-based dispatch.

Is there a way we can do this without having to add a ref to the map view?

Reply:
Discussed. @Archdoog to investigate moving the gesture recognizer callback handling to the map view itself via extensions to avoid this.

Reply:
Unfortunately, putting it on the map view had another issue. Reason being, we still needed a direct connection back to the UIViewRepresentable struct, which only the coordinator really has. So while getting the point/coordinate of the gesture within the mapView was easy directly on self, it was too disconnected from where we emit the context back up the SwiftUI stack.